lib/mailserver.js

```diff
@@ -258,14 +259,18 @@ mailServer.create = function(port, host, user, password) {
       smtp.on('authorizeUser', authorizeUser);
     }
   } else {
-    smtp = new SMTPServer({
+    smtpServerConfig = {
       incomingUser: user,
       incomingPassword: password,
       onAuth: authorizeUser,
       onData: handleDataStream,
       logger: false,
       disabledCommands: !!(user && password)?['STARTTLS']:['AUTH']
-    });
+    };
+
+    addHideExtensionOptions(smtpServerConfig, hideExtensions);
```

Comment:
This is a bit of a style thing, but I'd prefer a bit more functional way of writing this, perhaps something like:

```js
const smtpExtensionConfig = getValidExtensionOptions(hideExtensions)
smtpServerConfig = Object.assign({}, smtpServerConfig, smtpExtensionConfig)
```
That makes the code more testable. If we mutate the object passed by reference it isn't clear that it's changing the object. Can we refactor this?